File: src/FlowWorker.Shared/Entities/Session.cs
using System.Text.Json.Serialization;
using FlowWorker.Shared.Enums;

namespace FlowWorker.Shared.Entities;

/// <summary>
/// 会话实体
/// </summary>
public class Session
{
    /// <summary>
    /// 主键
    /// </summary>
    public Guid Id { get; set; }

    /// <summary>
    /// 会话标题
    /// </summary>
    public string Title { get; set; } = string.Empty;

    /// <summary>
    /// 会话类型（Single/Group）
    /// </summary>
    public SessionType Type { get; set; } = SessionType.Single;

    /// <summary>
    /// 创建者（User Member ID）
    /// </summary>
    public Guid CreatedBy { get; set; }

    /// <summary>
    /// 关联的 API 配置 ID（单聊时必填，群聊时为 null，因为每个 AI 成员有自己的 API 配置）
    /// </summary>
    public Guid? ApiConfigId { get; set; }

    /// <summary>
    /// 使用的模型（单聊时必填，群聊时为 null）
    /// </summary>
    public string Model { get; set; } = string.Empty;

    /// <summary>
    /// 系统提示词
    /// </summary>
    public string SystemPrompt { get; set; } = string.Empty;

    /// <summary>
    /// 生成温度
    /// </summary>
    public decimal Temperature { get; set; } = 0.7m;

    /// <summary>
    /// 最大 Token 数
    /// </summary>
    public int? MaxTokens { get; set; }

    /// <summary>
    /// 创建时间
    /// </summary>
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    /// <summary>
    /// 更新时间
    /// </summary>
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    /// <summary>
    /// 扩展元数据（JSON 格式）
    /// </summary>
    public string? Metadata { get; set; }

    /// <summary>
    /// 会话工作目录路径，限定AI工具的文件操作范围
    /// </summary>
    public string WorkingDirectory { get; set; } = string.Empty;

    /// <summary>
    /// 关联的 API 配置
    /// </summary>
    [JsonIgnore]
    public virtual ApiConfig? ApiConfig { get; set; }

    /// <summary>
    /// 关联的消息列表
    /// </summary>
    [JsonIgnore]
    public virtual ICollection<Message> Messages { get; set; } = new List<Message>();

    /// <summary>
    /// 关联的会话成员列表
    /// </summary>
    [JsonIgnore]
    public virtual ICollection<SessionMember> SessionMembers { get; set; } = new List<SessionMember>();
}

File: frontend/src/lib/types/session.ts
/**
 * 会话类型
 */
export enum SessionType {
    Single = 0,
    Group = 1
}

/**
 * 会话类型名称映射
 */
export const SessionTypeNames: Record<SessionType, string> = {
    [SessionType.Single]: '单聊',
    [SessionType.Group]: '群聊'
};

/**
 * 会话列表项
 */
export interface SessionListItemDto {
    id: string;
    title: string;
    type: SessionType;
    model: string;
    createdAt: string;
    updatedAt: string;
    messageCount: number;
    memberCount?: number;
}

/**
 * 会话详情
 */
export interface SessionDetailDto {
    id: string;
    title: string;
    type: SessionType;
    apiConfigId: string;
    apiConfigName: string;
    model: string;
    systemPrompt: string;
    temperature: number;
    maxTokens: number | null;
    createdAt: string;
    updatedAt: string;
    messages: Message[];
    members?: SessionMemberDto[];  // 后端返回的是 members，不是 participants
}

/**
 * 创建会话请求（单聊）
 */
export interface CreateSessionRequest {
    title: string;
    memberId: string;
    workingDirectory?: string;
}

/**
 * 更新会话请求
 */
export interface UpdateSessionRequest {
    title: string;
    apiConfigId: string;
    model: string;
    systemPrompt: string;
    temperature: number;
    maxTokens: number | null;
}

/**
 * 消息角色类型
 */
export enum MessageRole {
    System = 'system',
    User = 'user',
    Assistant = 'assistant',
    Tool = 'tool'
}

/**
 * 消息实体
 */
export interface Message {
    id: string;
    sessionId: string;
    role: MessageRole;
    content: string;
    tokens: number | null;
    model: string | null;
    createdAt: string;
    metadata: string | null;
}

// ========== 群聊相关类型 ==========

/**
 * 群聊会话详情
 */
export interface GroupSessionDetailDto {
    id: string;
    title: string;
    apiConfigId: string;
    apiConfigName: string;
    model: string;
    systemPrompt: string;
    temperature: number;
    maxTokens: number | null;
    createdAt: string;
    updatedAt: string;
    participants: ParticipantDto[];
    messages: GroupMessageDto[];
}

/**
 * 参与者信息
 */
export interface ParticipantDto {
    memberId: string;
    memberName: string;
    memberType: number | string;  // 后端返回数字：0 = User, 1 = AI
    roleName: string;
    order: number;
    isActive: boolean;
}

/**
 * 群聊消息
 */
export interface GroupMessageDto {
    id: string;
    sessionId: string;
    senderId: string;
    senderName: string;
    senderType: string;
    content: string;
    tokens: number | null;
    model: string | null;
    createdAt: string;
}

/**
 * 创建群聊会话请求
 */
export interface CreateGroupSessionRequest {
    title: string;
    createdBy: string;
    aiMemberIds: string[];
    systemPrompt?: string;
    workingDirectory?: string;
}

/**
 * 发送群聊消息请求
 */
export interface SendGroupMessageRequest {
    content: string;
    senderMemberId: string;
}

/**
 * 发送群聊消息结果
 */
export interface SendGroupMessageResult {
    messageId: string;
    responses: GroupMessageResponse[];
}

/**
 * 群聊消息响应
 */
export interface GroupMessageResponse {
    memberId: string;
    memberName: string;
    content: string;
    tokens: number | null;
}

/**
 * 群聊状态
 */
export interface GroupChatStateDto {
    isProcessing: boolean;
    currentMemberId: string | null;
    currentMemberName: string | null;
    processedCount: number;
    totalCount: number;
}

/**
 * 会话成员信息
 */
export interface SessionMemberDto {
    id: string;
    name: string;
    type: number | string;  // 后端返回数字：0 = User, 1 = AI
    avatar: string | null;
    status: string;
    roleName: string | null;
    roleDisplayName: string | null;
    joinedAt: string;
    isActive: boolean;
}
